src/graph.rs

```diff
@@ -111,56 +105,60 @@ impl<'tree> Graph<'tree> {
                 let ctx = self.1;
                 let mut seq = serializer.serialize_seq(Some(graph.graph_nodes.len()))?;
                 for (node_index, node) in graph.graph_nodes.iter().enumerate() {
-                    seq.serialize_element(&JSONNode(node_index, node, ctx))?;
+                    seq.serialize_element(&InContext(&(node_index, node), ctx))?;
                 }
                 seq.end()
             }
         }
 
-        impl<'a> ser::Serialize for JSONNode<'a> {
+        impl<'a> ser::Serialize for InContext<'a, (usize, &'a GraphNode)> {
             fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
             where
                 S: serde::Serializer,
             {
-                let node_index = self.0;
-                let node = self.1;
-                let ctx = self.2;
+                let node_index = self.0 .0;
+                let node = self.0 .1;
+                let ctx = self.1;
                 // serializing as a map instead of a struct so we don't have to encode a struct name
                 let mut map = serializer.serialize_map(None)?;
                 map.serialize_entry("index", &node_index)?;
```

Comment:
This corresponds exactly to the position of this entry in the enclosing list, right? Is it useful to have this explicitly as a property as well, you think? Or do you consider it purely accidental that this index happens to coincide with the position, and you should always search the list for the entry with the correct index (which in that case should perhaps be named `id`)?

Reply:
AFAIK it does correspond thus, but I'm considering this as future-proofing against that decision changing. `display_with` makes the same decision, which is probably what I was cribbing  from.

`id` is definitely a better name, and probably something similar applies to `Edge`s.